src/components/metadata-modal/metadata-modal.js

```diff
@@ -33,19 +35,33 @@ const PlotlyModal = ({ metadata, onToggle, visible }) => {
           <span className="pipeline-plot-modal__title">{metadata.name}</span>
         </div>
       </div>
-      <PlotlyChart
-        data={metadata.plot.data}
-        layout={metadata.plot.layout}
-        view="modal"
-      />
+      {metadata?.plot ? (
+        <PlotlyChart
+          data={metadata.plot.data}
+          layout={metadata.plot.layout}
+          view="modal"
+        />
+      ) : (
+        <div className="pipeline-matplolib-chart">
+          <div className="pipeline-metadata__plot-image-container">
+            <img
+              alt="Matplotlib rendering"
+              className="pipeline-metadata__plot-image--expanded"
+              src={`data:image/png;base64,${metadata.image}`}
+            />
+          </div>
+        </div>
+      )}
       <div className="pipeline-plot-modal__bottom">
         <button
           className="pipeline-plot-modal__collapse-plot"
           onClick={onCollapsePlotClick}
         >
           <CollapseIcon className="pipeline-plot-modal__collapse-plot-icon"></CollapseIcon>
           <span className="pipeline-plot-modal__collapse-plot-text">
-            Collapse Plotly Visualization
+            {metadata?.plot
```

Comment:
Is it possible that a node with a `dataset_type` of plotly or matplotlib won't contain and `image` or a `plot`? If it is, I think we'll still need to check for either `image` or `plot`. In fact, this did happen to me yesterday. I added in my new node but didn't do `kedro run`, so the image wasn't there but the node on the flowchart was present, as was the metadata panel.

I agree with everything else. The logic isn't identical and should be. I'll update it.

Reply:
Ah yes, that is a very good point. If you haven't done a `kedro run` yet then `plot`/`image` won't be defined.

In this case I guess there's two bits of logic we need:
* something to check the `dataset_type` (probably using `shortTypeMapping`? Something like `switch shortTypeMapping[dataset_type]` or whatever)
* something to check whether the dataset actually has `image` or `plot` defined

Reply:
Also, curious: is there any reason these explicitly cast to `Boolean`? Seems like some things in metadata.js are and some aren't:
```
  const hasCode = Boolean(metadata?.code);
  const isTranscoded = Boolean(metadata?.originalType);
  const showCodePanel = visible && visibleCode && hasCode;
  const showCodeSwitch = hasCode;
```

Reply:
Good catch. Added those two to this, as suggested: 

```jsx
if (!visible.metadataModal || (!metadata?.plot && !metadata?.image)) {
```

The first two are cast as such because they're being converted from strings. The last two don't need it, since they're created from booleans they'll be that type, too.